src/Ansatz/Chain.jl

```diff
@@ -30,52 +30,98 @@ function Chain(tn::TensorNetwork, sites, args...; kwargs...)
     Chain(Quantum(tn, sites), args...; kwargs...)
 end
 
-function Chain(::State, boundary::Periodic, arrays::Vector{<:AbstractArray})
+defaultorder(::Type{Chain}, ::State) = (:o, :l, :r)
+defaultorder(::Type{Chain}, ::Operator) = (:o, :i, :l, :r)
+
+function Chain(::State, boundary::Periodic, arrays::Vector{<:AbstractArray}; order = defaultorder(Chain, State()))
     @assert all(==(3) ∘ ndims, arrays) "All arrays must have 3 dimensions"
+    issetequal(order, defaultorder(Chain, State())) ||
+        throw(ArgumentError("order must be a permutation of $(String.(defaultorder(State())))"))
 
     n = length(arrays)
     symbols = [nextindex() for _ in 1:2n]
 
     _tensors = map(enumerate(arrays)) do (i, array)
-        Tensor(array, [symbols[i], symbols[n+mod1(i - 1, n)], symbols[n+mod1(i, n)]])
+        inds = map(order) do dir
+            if dir == :o
+                symbols[i]
+            elseif dir == :r
+                symbols[n+mod1(i, n)]
+            elseif dir == :l
+                symbols[n+mod1(i - 1, n)]
+            else
+                throw(ArgumentError("Invalid direction: $dir"))
+            end
+        end
+        Tensor(array, inds)
     end
 
     sitemap = Dict(Site(i) => symbols[i] for i in 1:n)
 
     Chain(Quantum(TensorNetwork(_tensors), sitemap), boundary)
 end
 
-function Chain(::State, boundary::Open, arrays::Vector{<:AbstractArray})
+function Chain(::State, boundary::Open, arrays::Vector{<:AbstractArray}; order = defaultorder(Chain, State()))
     @assert ndims(arrays[1]) == 2 "First array must have 2 dimensions"
     @assert all(==(3) ∘ ndims, arrays[2:end-1]) "All arrays must have 3 dimensions"
     @assert ndims(arrays[end]) == 2 "Last array must have 2 dimensions"
+    issetequal(order, defaultorder(Chain, State())) ||
+        throw(ArgumentError("order must be a permutation of $(String.(defaultorder(Chain, State())))"))
 
     n = length(arrays)
-    symbols = [nextindex() for _ in 1:2n-1]
+    symbols = [nextindex() for _ in 1:2n]
 
     _tensors = map(enumerate(arrays)) do (i, array)
         if i == 1
-            Tensor(array, [symbols[1], symbols[1+n]])
+            _order = filter(x -> x != :l, order)
         elseif i == n
-            Tensor(array, [symbols[n], symbols[n+mod1(n - 1, n)]])
+            _order = filter(x -> x != :r, order)
         else
-            Tensor(array, [symbols[i], symbols[n+mod1(i - 1, n)], symbols[n+mod1(i, n)]])
+            _order = order
```

Comment:
Maybe prettier like this?
```suggestion
         local order = if i == 1
             filter(x -> x != :l, order)
         elseif i == n
             filter(x -> x != :r, order)
         else
             order
         end
```